runtime/registry.go

```diff
@@ -16,7 +17,17 @@ func (r *Runtime) FindInstances(ctx context.Context) ([]*drivers.Instance, error
 }
 
 func (r *Runtime) FindInstance(ctx context.Context, instanceID string) (*drivers.Instance, error) {
-	return r.Registry().FindInstance(ctx, instanceID)
+	instance, err := r.Registry().FindInstance(ctx, instanceID)
+	emitter.AddDimsToContext(ctx, *emitter.String("instance_id", instanceID))
+
+	if err != nil && instance != nil {
+		var usageDims []emitter.Dim
+		for k, v := range instance.Labels {
+			usageDims = append(usageDims, *emitter.String(k, v))
+		}
+		emitter.AddDimsToContext(ctx, usageDims...)
+	}
+	return instance, err
 }
```

Comment:
`FindInstance` may be called for a variety of reasons – even if it's *likely* to only be called for a single instance on instance-scoped API calls, that dependency is too implicit for us to rely on it for tracking.

I would consider a more explicit approach, similar to logging. Perhaps having a `usageClient.WithDims(...)` wrapper could help (not sure).

Reply:
Seems like every request with an instance id restores the instance from a DB thus I updated the logic so that it explicitly stores the instance id into the context and at the time FindInstance is called, the context is populated by the instance labels. This keeps the same number of requests to the metastore.